api/v1beta1/rollout_types.go

```diff
@@ -101,17 +81,17 @@ type RolloutStrategy struct {
 	// Default value is false
 	Paused bool `json:"paused,omitempty"`
 	// +optional
+	// Traffic Graying, include canary release, A/B Testing release
 	Canary *CanaryStrategy `json:"canary,omitempty"`
-}
-
-// CanaryStrategy defines parameters for a Replica Based Canary
-type CanaryStrategy struct {
-	// Steps define the order of phases to execute release in batches(20%, 40%, 60%, 80%, 100%)
+	// Batches is batch release capacity, and does not contain traffic gray
+	//Users can specify their batch plan in this field, such as:
+	// batches:
+	// - replicas: 1  # batches 0
+	// - replicas: 2  # batches 1
+	// - replicas: 5  # batches 2
+	// Not that these canaryReplicas should be a non-decreasing sequence.
 	// +optional
-	Steps []CanaryStep `json:"steps,omitempty"`
-	// TrafficRoutings hosts all the supported service meshes supported to enable more fine-grained traffic routing
-	// and current only support one TrafficRouting
-	TrafficRoutings []v1alpha1.TrafficRoutingRef `json:"trafficRoutings,omitempty"`
+	Batches []BatchStrategy `json:"batches,omitempty"`
```

Comment:
Batches should be pointer to BatchStrategy, and in BatchStrategy , one can define replicas for each batches (maybe a field called BatchStep), and other strategy configuration